libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI.m

```diff
@@ -176,10 +179,14 @@ - (void)send:(SFRestRequest *)request delegate:(id<SFRestDelegate>)delegate shou
     if (nil != delegate) {
         request.delegate = delegate;
     }
-    [self.activeRequests addObject:request];
 
     // If there are no demonstrable auth credentials, login before sending.
     SFUserAccount *user = [SFUserAccountManager sharedInstance].currentUser;
+
+    // Adds this request to the list of active requests if it's not already on the list.
+    if (![self.activeRequests containsObject:request]) {
```

Comment:
Done in latest commit [here](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/2152/commits/f70d0b4bc685edfdd12e8f2cd85a1ec910c1e199).